footprint/src/main/java/com/umc/footprint/src/badge/BadgeService.java

```diff
@@ -258,4 +251,53 @@ public BadgeDateInfo modifyRepBadge(String userId, int badgeIdx) throws BaseExce
             throw new BaseException(DATABASE_ERROR);
         }
     }
+
+    @Transactional
+    public List<Integer> getAcquiredBadgeIdxList(int userIdx, Double totalDistance, Integer totalWalkCount) {
+        // 현재 누적 거리, 횟수로 얻을 수 있는 뱃지 계산 - 획득한 뱃지가 없을 때 어떤 동작을 해야 하지?
+        TotalDistanceStatus distanceStatus = TotalDistanceStatus.getDistanceBadge(totalDistance).orElseThrow();
+        TotalRecordStatus recordStatus = TotalRecordStatus.getWalkCountBadge(totalWalkCount).orElseThrow();
+
+        List<Integer> acquiredBadgeIdxList = new ArrayList<>();
+
+        // 현재 갖고 있는 뱃지 조회
+        List<Integer> beforeSavingWalkBadgeList = userBadgeRepository.findAllByUserIdxAndStatus(userIdx, "ACTIVE")
+                .map(userBadgeList -> userBadgeList.stream()
+                        .map(UserBadge::getBadgeIdx)
+                        .sorted()
+                        .collect(Collectors.toList()))
+                .orElse(Collections.emptyList());
+
+        if (beforeSavingWalkBadgeList.isEmpty()) { // 발자국 스타터 부여
+            acquiredBadgeIdxList.add(1);
+        }
+
+        int originMaxDistanceBadgeIdx = 1; // 원래 갖고 있던 뱃지(2~5)의 가장 큰 값
+        int originMaxRecordBadgeIdx = 1; // 원래 갖고 있던 뱃지(6~8)의 가장 큰 값
+
+        for (Integer originBadgeIdx : beforeSavingWalkBadgeList) {
+            if (originBadgeIdx >= TotalDistanceStatus.minBadgeIdx() && originBadgeIdx <= TotalDistanceStatus.maxBadgeIdx()) {
+                originMaxDistanceBadgeIdx = originBadgeIdx;
+            }
+            if (originBadgeIdx >= TotalRecordStatus.minBadgeIdx() && originBadgeIdx <= TotalRecordStatus.maxBadgeIdx()) {
+                originMaxRecordBadgeIdx = originBadgeIdx;
+            }
+        }
+
+        // 기록 관련 뱃지 저장 - 6~8
+        if(distanceStatus.getBadgeIdx() > originMaxDistanceBadgeIdx) {
+            for (int i = originMaxDistanceBadgeIdx + 1; i <= distanceStatus.getBadgeIdx(); i++) {
+                acquiredBadgeIdxList.add(i);
+            }
+        }
+
+        //거리 뱃지 확인 로직 - 2~5
+        if(recordStatus.getBadgeIdx() > originMaxRecordBadgeIdx) {
+            for (int i = originMaxRecordBadgeIdx + 1; i <= recordStatus.getBadgeIdx(); i++) {
+                acquiredBadgeIdxList.add(i);
+            }
+        }
+
+        return acquiredBadgeIdxList;
```

Comment:
```suggestion
        return Collections.unmodifiableList(acquiredBadgeIdxList);
```
메서드 밖에서 리스트에 추가 못하게 불변 리스트로 반환하는 것도 좋을 것 같아요